core/state_transition.go

```diff
@@ -252,3 +262,105 @@ func (st *StateTransition) refundGas() {
 func (st *StateTransition) gasUsed() uint64 {
 	return st.initialGas - st.gas
 }
+
+// StakingTransitionDb will transition the state by applying the staking message and
+// returning the result including the used gas. It returns an error if failed.
+// It is used for staking transaction only
+func (st *StateTransition) StakingTransitionDb() (usedGas uint64, err error) {
+	if err = st.preCheck(); err != nil {
+		return 0, err
+	}
+	msg := st.msg
+	sender := vm.AccountRef(msg.From())
+	homestead := st.evm.ChainConfig().IsS3(st.evm.EpochNumber) // s3 includes homestead
+
+	// Pay intrinsic gas
+	gas, err := IntrinsicGas(st.data, false, homestead)
```

Comment:
This should be a new Intrinsic gas formula for staking. Add a todo please.

Reply:
added